src/insights_client/tests/test_motd.py

```diff
@@ -0,0 +1,116 @@
+import os
+import tempfile
+import unittest
+import unittest.mock
+
+import pytest
+
+import insights_client
+
+
+class MockFS:
+    """Mock filesystem.
+
+    Ensures that the files required for the MOTD manipulation are created
+    in a temporary directory instead of working with actual system files.
+    """
+
+    def __init__(self):
+        self.dir = tempfile.TemporaryDirectory()
+
+        # /etc/insights-client/insights-client.motd
+        self.dir_src = unittest.mock.patch(
+            "insights_client.MOTD_SRC", f"{self.dir.name}/src"
+        )
+        self.dir_src.start()
+
+        # /etc/motd.d/insights-client
+        self.dir_motd = unittest.mock.patch(
+            "insights_client.MOTD_FILE", f"{self.dir.name}/motd"
+        )
+        self.dir_motd.start()
+
+        # /etc/insights-client/.registered
+        self.dir_reg = unittest.mock.patch(
+            "insights_client.REGISTERED_FILE", f"{self.dir.name}/reg"
+        )
+        self.dir_reg.start()
+
+        # /etc/insights-client/.unregistered
+        self.dir_unreg = unittest.mock.patch(
+            "insights_client.UNREGISTERED_FILE", f"{self.dir.name}/unreg"
+        )
+        self.dir_unreg.start()
```

Comment:
There is a lot of repetition, which can be replaces by a for-cycle.

```suggestion
        DIRS = [
            ("insights_client.MOTD_SRC", "/src"),  # /etc/insights-client/insights-client.motd
            ("insights_client.MOTD_FILE", "/motd"),  # /etc/motd.d/insights-client
            ("insights_client.REGISTERED_FILE", "/reg"),  # /etc/insights-client/.registered
            ("insights_client.UNREGISTERED_FILE", "/unreg"),  # /etc/insights-client/.unregistered
        ]
        
        self.dirs = {}
        for target, value in DIRS:
            self.dirs[target] = unittest.mock.patch(target, f"{self.dir.name}{value}"
            self.dirs[target].start()
```